sycl-fusion/jit-compiler/lib/translation/SPIRVLLVMTranslation.cpp

```diff
@@ -0,0 +1,192 @@
+//==---------------------- SPIRVLLVMTranslation.cpp ------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "SPIRVLLVMTranslation.h"
+
+#include "Kernel.h"
+#include "LLVMSPIRVLib.h"
+#include "helper/ErrorHandling.h"
+#include "llvm/ADT/DenseSet.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Linker/Linker.h"
+#include "llvm/Support/raw_ostream.h"
+#include <iostream>
+#include <sstream>
+
+using namespace jit_compiler;
+using namespace jit_compiler::translation;
+using namespace llvm;
+
+void SPIRVLLVMTranslator::getAttributeValues(std::vector<std::string> &Values,
+                                             MDNode *MD, size_t NumValues) {
+  assert(MD->getNumOperands() == NumValues && "Incorrect number of values");
+  for (const auto &MDOp : MD->operands()) {
+    auto *ConstantMD = dyn_cast<ConstantAsMetadata>(MDOp);
+    assert(ConstantMD && "Metadata not constant");
```

Comment:
```suggestion
    auto *ConstantMD = cast<ConstantAsMetadata>(MDOp);
```

`cast` already has embedded `assert` within it.